plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -6910,6 +6910,211 @@ public void testNoRetryWhenMetadataFileInvalid()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
```

Comment:
Please add more tests about other statement types. For instance, `optimize` procedure is affected, but this PR doesn't have the test coverage.

Reply:
This is a really great point. I have tested the other statement types, and eventually decided to restrict the scope of the config flag to SELECT statements only (which is the main use case anyway). Updated the config description in the doc to reflect that. Other statement types such as MERGE and ANALYZE which also carry out a table scan cannot even have a where clause and hence would fail with this config turned on in all cases. OPTIMIZE could take a WHERE clause but often it's more convenient to run on the entire table. So all in all, it looks to be much cleaner to just use the config for SELECTs.

Reply:
Hive and Delta Lake connector enforces the property for non-SELECT statements as well. It would be nice to keep the same behavior in Iceberg connector in my opinion.

Reply:
@ebyhr Sure, makes sense to be consistent. I have updated the unit test to cover merge/update/delete/optimize/analyze. To support ANALYZE statements which cannot take any predicates, I've added a new analyzeColumns field to the iceberg table handle to detect that the statement type is ANALYZE. It is done similarly by Delta Lake. Please let me know if it looks all good now to go in. Thank you!